src/lib/utils.ts

```diff
@@ -48,6 +48,23 @@ export const isIssuedTimeWithinWindow = (expiresAtMs: number, expiresInMs: numbe
   return Date.now() - issuedAt < windowMs;
 }
 
+export class TranslatableError extends Error {
+  translationKey: string;
```

Comment:
I like this concept. Maybe one improvement could be to strongly type out these keys? That way we would remember to check the translation files to make sure we have a translate for them. Something like: 
``` typescript
translationKey: 'errorToken' | 'errorStateParam' | 'errorParse' ...
```

Reply:
This is a good idea, but my only reservation would be in the event the list of possible errors grows large, we'd have to have a key for every one of them here that we want to translate as well as update the `i18n.json` files.